iac-extensions/arm/src/test/java/org/sonar/iac/arm/parser/ResourceDeclarationTest.java

```diff
@@ -0,0 +1,201 @@
+    assertThat(properties.get(0).value().value()).isEqualTo("random location");
+    IacCommonAssertions.assertThat(properties.get(0).textRange()).hasRange(7, 6, 7, 35);
+  }
+
+  @Test
+  void checkChildren() throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+    String code = code("{",
+      "  \"resources\": [",
+      "    {",
+      "      \"type\": \"Microsoft.Kusto/clusters\",",
+      "      \"apiVersion\": \"2022-12-29\",",
+      "      \"name\": \"myResource\",",
+      "      \"location\": \"random location\",",
+      "    }",
+      "  ]",
+      "}");
+    File tree = (File) parser.parse(code, null);
+    ResourceDeclaration resourceDeclaration = (ResourceDeclaration) tree.statements().get(0);
+
+    List<Tree> children = resourceDeclaration.children();
+    assertThat(children).hasSize(8);
+
+    checkElement(children.get(0), ArmTree.Kind.IDENTIFIER, "name", 6, 6, 6, 12);
+    checkElement(children.get(1), ArmTree.Kind.EXPRESSION, "myResource", 6, 14, 6, 26);
+    checkElement(children.get(2), ArmTree.Kind.IDENTIFIER, "apiVersion", 5, 6, 5, 18);
+    checkElement(children.get(3), ArmTree.Kind.EXPRESSION, "2022-12-29", 5, 20, 5, 32);
+    checkElement(children.get(4), ArmTree.Kind.IDENTIFIER, "type", 4, 6, 4, 12);
+    checkElement(children.get(5), ArmTree.Kind.EXPRESSION, "Microsoft.Kusto/clusters", 4, 14, 4, 40);
+    checkElement(children.get(6), ArmTree.Kind.IDENTIFIER, "location", 7, 6, 7, 16);
+    checkElement(children.get(7), ArmTree.Kind.EXPRESSION, "random location", 7, 18, 7, 35);
+  }
+
+  void checkElement(Tree element, ArmTree.Kind kind, String value, int startLine, int startLineOffset, int endLine, int endLineOffset)
+    throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    assertThat(element).isInstanceOf(ArmTree.class);
+    ArmTree armTree = (ArmTree) element;
+    assertThat(armTree.is(kind)).isTrue();
+
+    Method method = element.getClass().getMethod("value");
+    String elementValue = (String) method.invoke(element);
+    assertThat(elementValue).isEqualTo(value);
+
+    assertThat(armTree.children()).isEmpty();
+    IacCommonAssertions.assertThat(armTree.textRange()).hasRange(startLine, startLineOffset, endLine, endLineOffset);
+  }
+
+  @Test
+  void invalidNameExpression() {
+    String code = code("{",
+      "  \"resources\": [",
+      "    {",
+      "      \"type\": \"Microsoft.Kusto/clusters\",",
+      "      \"apiVersion\": \"2022-12-29\",",
+      "      \"name\": [],",
+      "      \"location\": \"random location\",",
+      "    }",
+      "  ]",
+      "}");
+    assertThatThrownBy(() -> parser.parse(code, null))
+      .isInstanceOf(ParseException.class)
+      .hasMessage("Expecting ScalarTree to convert to Expression, got SequenceTreeImpl");
+  }
+
+  @Test
+  void shouldThrowParseExceptionOnIncompleteResource() {
+    String[] usecases = new String[] {
```

Comment:
It looks like a good candidate for Parameterized Tests: https://www.baeldung.com/parameterized-tests-junit-5